### ISSUE TITLE
Some issues with the react app

### ISSUE THREAD

**Issue by keystroke3** [Mathewsmusukuma/foto-dino-locations#1]
Greetings Mathew,  
I am a member of the FotoDino backend team and I am responsible for looking at the apps submitted for the front end job.
First I would like to thank you for applying for the position. It shows great initiative on your part. One of the main reasons for us creating this test was to weed out the lazy and arrogant applicants. You do not seem to be one of them.

### Issues
That being said, I looked into and fixed the CORS issue that you were facing. The API should now work as expected. I also looked at the app, and noticed you only have the `Cities` implemented. I assume that is a product of the CORS error. Now that that has been fixed, I believe you should be able to complete the `Locations` section.

I also noticed a few problems in your current implementation of `Cities`.
#### Search/Filter
The search part of the api is supposed to allow you to isolate specific cities matching a particular criterion. It was expected that the search function should be able to either make a call through the `search` part of the api backend and fetch cities based on the query. You could also, use tools in React to perform a fuzzy search on the `name` or `code` fields. Your implementation does not seem to do that.
![image](https://user-images.githubusercontent.com/37796862/133906329-608567bd-b989-465e-a8d9-96286ac10ce7.png)
The search is supposed to return any and all cities matching the give query, by looking into names *and* codes. The filter option is supposed to use a specific field, `name` *or* `code` to find one specific city. In your implementation, `filter by` seems to be performing sorting instead of filtering. You should keep the `filter by` drop down, but rename it to `sort by`. Then add another smaller dropdown next to to the search bar with `all fields`, for search and `name`, `code` for filtering.

#### City creation
Another issue that I found, was in the `Create City` form popup. The locations field is automatically generated by the api, and is a list of the ids of all locations ( hotels) in that city. Therefor the field `Enter Locations` is not needed. The other fields have no labels and is easy to fill the wrong information in the wrong box. They should have labels.
![issues with cities](https://user-images.githubusercontent.com/37796862/133906532-95882fbd-0f4a-4091-8a63-2cdfa17dc3e8.png)

#### Edit City
The `Edit City` pop up form should allow the user to modify the city code. It currently only allows the modification of the city name. Also, the `Update` button does not seem to do anything
![missing city code edit](https://user-images.githubusercontent.com/37796862/133906621-1e3cd0dc-bbba-453a-8148-a205867e065d.png)

#### Detailed view
I assume also that this was not completed due to the CORS error. There should be a detailed view of each city, that shows all the locations in the city, and when the last update was, etc. In this detailed view,  is where we can have the option of updating the city properties and also selecting a paritcular location to view details, and deleting the city. To make this, you can use the `locations` field list of locaiton ids to location details from api or you can query the `/locations` api endpoint with a filter for the city id. Both of these approaches will give you the same result, but one is faster and simpler than the other. I will let you figure out which is which.

#### Page Layout
Due to the shear number of items to be displayed, it was expected that you would go with a grid layout, and implement pagination. The `Cities` and `Locations` overviews are supposed to be 'bird's eye view', therefor each item should take the least amount of space possible.

### Closing Thoughts
You have a very nice implementation and would like to see what else you can do. I have not looked at the code, as that will be done once the application is complete and you are sure you have finished everything. Implement the `Locations` view putting all I have stated above into consideration, and make the necessary changes to the current `City` view. Please note that this is not the final review, unless you choose not to continue, and we will reserve our judgement until after you complete all the tasks listed in the job offering. You have 48 more hours.
Thank You.
Ted

**Comment by Mathewsmusukuma** [comment]
Hi @keystroke3 , thank you for the great feedback, I appreciate it. Sorry that I did not respond on time, as I never got a notification about the issue until Ivo told me about. I am working on the fixes and implementations. Thank you.
